yarn.lock

```diff
@@ -14781,20 +14781,20 @@ mkdirp-promise@^5.0.1:
   dependencies:
     mkdirp "*"
 
-mkdirp@*, mkdirp@0.5.1, "mkdirp@>=0.5 0", mkdirp@^0.5.0, mkdirp@^0.5.1, mkdirp@~0.5.1:
-  version "0.5.1"
-  resolved "https://registry.yarnpkg.com/mkdirp/-/mkdirp-0.5.1.tgz#30057438eac6cf7f8c4767f38648d6697d75c903"
-  integrity sha1-MAV0OOrGz3+MR2fzhkjWaX11yQM=
-  dependencies:
-    minimist "0.0.8"
-
-mkdirp@^0.5.4:
+mkdirp@*, "mkdirp@>=0.5 0", mkdirp@^0.5.0, mkdirp@^0.5.1, mkdirp@^0.5.4, mkdirp@~0.5.1:
   version "0.5.5"
   resolved "https://registry.yarnpkg.com/mkdirp/-/mkdirp-0.5.5.tgz#d91cefd62d1436ca0f41620e251288d420099def"
   integrity sha512-NKmAlESf6jMGym1++R0Ra7wvhV+wFW63FaSOFPwRahvea0gMUcGUhVeAg/0BC0wiv9ih5NYPB1Wn1UEI1/L+xQ==
   dependencies:
     minimist "^1.2.5"
 
+mkdirp@0.5.1:
+  version "0.5.1"
+  resolved "https://registry.yarnpkg.com/mkdirp/-/mkdirp-0.5.1.tgz#30057438eac6cf7f8c4767f38648d6697d75c903"
+  integrity sha1-MAV0OOrGz3+MR2fzhkjWaX11yQM=
+  dependencies:
+    minimist "0.0.8"
+
```

Comment:
Okay... so just to make sure I understand what this diff means :)

What we're changing is moving the exact version dependency of the lib at 0.5.1 to its own entry, and then everything else that is allowed to resolve to 0.5.5 according to semver logic will get upgraded to that?

Is this... as safe change? I mean I know it's supposed to be from a semver perspective and I guess we should trust the authors of the libraries that they're appropriately marking lib versions (is that safe?) 🤔

Reply:
> just to make sure I understand what this diff means

You got it 100% correct!

> Is this... as safe change?

As you said, we are trusting that devs follow semver, and when they depend on a version range they understand what they are doing. 

yarn says this is unsafe to do automatically because there will be a code path that now will get a different version of the lib, which is true. They want humans to verify this changes, which I guess it makes sense.

Personally, given that these changes are mostly on devDependencies, there are not big version bumps, CI is still passing and there are no changes in size in the bundles (which I assume as a proxy for 'there are no changes in the bundles) and we were already using `mkdir 0.5.5` in some places, I'm fairly confident these changes are harmless.

For extra safety I could have linked to the library repo so the reviewer can see the changelog between those two versions. But in the case of `mkdirp`.... well, not a great changelog: https://github.com/isaacs/node-mkdirp/blob/master/CHANGELOG.md

So it is safe-ish :)

Reply:
Right on! Thanks for the confirmation and explanation 😊 Makes total sense to me. I also noticed that they were mostly dev deps so wasn't too worried about it 😁